### PR TITLE
[lang] Texture image load store support

### PR DESCRIPTION
**Pull request by bobcao3**
Related issue = #5109

### CODE REVIEW

File: taichi/ir/statements.h [hunk 1]
```diff
@@ -1453,13 +1453,39 @@ class TexturePtrStmt : public Stmt {
  public:
   Stmt *arg_load_stmt{nullptr};
   int dimensions{2};
+  bool is_storage{false};
```
Comment (k-ye) [inline taichi/ir/statements.h:1456]: nit: Put this to the end so that we avoid paddings

File: taichi/ir/stmt_op_types.h [hunk 2]
```diff
@@ -84,7 +84,13 @@ enum class SNodeOpType : int {
 
 std::string snode_op_type_name(SNodeOpType type);
 
-enum class TextureOpType : int { sample_lod, fetch_texel, undefined };
+enum class TextureOpType : int {
+  sample_lod,
+  fetch_texel,
+  load,
+  store,
+  undefined
```
Comment (k-ye) [inline taichi/ir/stmt_op_types.h:92]: I think the convention is to put `undefined` in the first place? (Also we should rename this to `kXx` style later)

File: taichi/rhi/device.h [hunk 3]
```diff
@@ -148,6 +148,13 @@ class ResourceBinder {
     TI_NOT_IMPLEMENTED
   }
 
+  virtual void rw_image(uint32_t set,
+                        uint32_t binding,
+                        DeviceAllocation alloc,
+                        int lod) {
+    TI_NOT_IMPLEMENTED
```
Comment (k-ye) [inline taichi/rhi/device.h:155]: This can be `=0` since you have overriden each device impl.

File: python/taichi/lang/_texture.py [hunk 4]
```diff
@@ -61,6 +61,53 @@ def fetch(self, index, lod):
         return ti.Vector([r, g, b, a])
 
 
+class RWTextureAccessor:
```
Comment (k-ye) [inline python/taichi/lang/_texture.py:64]: @strongoier @jim19930609 any concerns here for C++ vec types?

File: taichi/rhi/device.h (hunk 3 above)
Comment (bobcao3) [reply taichi/rhi/device.h:155]: This is intended for devices that do not support this doesn't need to have a override functions

File: taichi/inc/frontend_statements.inc.h [hunk 5]
```diff
@@ -13,3 +13,4 @@ PER_STATEMENT(FrontendSNodeOpStmt)  // activate, deactivate, append, clear
 PER_STATEMENT(FrontendAssertStmt)
 PER_STATEMENT(FrontendFuncDefStmt)
 PER_STATEMENT(FrontendReturnStmt)
+PER_STATEMENT(FrontendTextureOpStmt)
```
[the remark this answers is not in the document]
Comment (bobcao3) [reply taichi/inc/frontend_statements.inc.h:16]: How do we make sure that image store works? It does not return

[the remark this answers is not in the document]
Comment (strongoier) [reply taichi/inc/frontend_statements.inc.h:16]: Wrap the expr with `expr_init()`, just like https://github.com/taichi-dev/taichi/blob/31f46dd551fdd77a5e22313d0c2dc4d8942916b4/python/taichi/lang/impl.py#L929-L932

**Review by k-ye — APPROVED** [review]
LGTM, thanks!

**Pull request merged**